internal/vehicle_template.go

```diff
@@ -28,11 +29,22 @@ func NewVehicleTemplates(logger zerolog.Logger, vsd gateways.VehicleSignalDecodi
 
 // GetTemplateSettings checks for any new template settings and if so updates the local settings, returning the latest
 // settings. Logs if encounters errors along the way. Continues and gets local settings if can't get anything from remote. Errors if can't get anything useful.
-func (vt *vehicleTemplates) GetTemplateSettings(addr *common.Address) (*models.TemplatePIDs, *models.TemplateDeviceSettings, error) {
+func (vt *vehicleTemplates) GetTemplateSettings(addr *common.Address, vl loggers.VINLogger, unitID uuid.UUID) (*models.TemplatePIDs, *models.TemplateDeviceSettings, error) {
 	vinConfig, err := vt.lss.ReadVINConfig() // should this be more of VIN info? stored VIN info
 	if err != nil {
 		vt.logger.Err(err).Msg("could not read local settings for stored VIN, continuing")
 	}
+
+	// if we can't find VIN from template file, then we need to query it from OBD
+	vinResp, err := vl.GetVIN(unitID, nil)
```

Comment:
wrap this in something lik `if vinConfig.vin == ""` so we only query OBD vin if we didn't have it stored.